From 24b023eed7e7696b01ee9edc365d57378e7820f5 Mon Sep 17 00:00:00 2001
From: Kevin Chen <kevin_chen@aspeedtech.com>
Date: Thu, 11 Sep 2025 19:12:08 +0800
Subject: [PATCH 25/32] arm/riscv: aspeed: Add reserved-memory nodes for ATF
 and OP-TEE

Introduce reserved-memory nodes to define memory regions for Trusted
Firmware-A (ATF) and OP-TEE in ARM and RISC-V device trees. These regions
are marked with `no-map` to prevent mapping by the OS.

Change-Id: I13dd20d0aed4c11c0a0b2326836c0cf655fa8629
Signed-off-by: Kevin Chen <kevin_chen@aspeedtech.com>
(cherry picked from commit 2fc8d13acaf7f0b651eedad4cdc95f5c0fe5eae0)
Signed-off-by: Vasanthakumar K <vasanthakumark@ami.com>
---
 arch/arm/dts/ast2700.dtsi   | 16 ++++++++++++++++
 arch/riscv/dts/ast2700.dtsi | 16 ++++++++++++++++
 2 files changed, 32 insertions(+)

diff --git a/arch/arm/dts/ast2700.dtsi b/arch/arm/dts/ast2700.dtsi
index 070037d6e7..cd014d9835 100644
--- a/arch/arm/dts/ast2700.dtsi
+++ b/arch/arm/dts/ast2700.dtsi
@@ -92,6 +92,22 @@
 			<0 0x40440000 0 0x1000>;		//GICC
 	};
 
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		atf: trusted-firmware-a@430000000 {
+			reg = <0x4 0x30000000 0x0 0x80000>;
+			no-map;
+		};
+
+		optee_core: optee-core@430080000 {
+			reg = <0x4 0x30080000 0x0 0x1000000>;
+			no-map;
+		};
+	};
+
 	timer {
 		compatible = "arm,armv8-timer";
 		interrupt-parent = <&gic>;
diff --git a/arch/riscv/dts/ast2700.dtsi b/arch/riscv/dts/ast2700.dtsi
index 437ff14661..45764bcd20 100644
--- a/arch/riscv/dts/ast2700.dtsi
+++ b/arch/riscv/dts/ast2700.dtsi
@@ -29,6 +29,22 @@
 		reg = <0x80000000 0x80000000>;
 	};
 
+	reserved-memory {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+
+		atf: trusted-firmware-a@b0000000 {
+			reg = <0xb0000000 0x80000>;
+			no-map;
+		};
+
+		optee_core: optee-core@b0080000 {
+			reg = <0xb0080000 0x1000000>;
+			no-map;
+		};
+	};
+
 	ast_loader {
 		compatible = "aspeed,ast_loader";
 
-- 
2.34.1

